cfme/infrastructure/resource_pool.py

```diff
@@ -96,16 +93,39 @@ def _on_detail_page(self):
     @property
     def exists(self):
         try:
-            sel.force_navigate('infrastructure_resource_pool', context=self._get_context())
-            quad = Quadicon(self.name, 'resource_pool')
+            navigate_to(self, 'All')
+            quad = Quadicon(self.name, self.quad_name)
             if sel.is_displayed(quad):
                 return True
         except sel.NoSuchElementException:
             return False
 
 
+@navigator.register(ResourcePool, 'All')
+class All(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance', 'LoggedIn')
+
+    def step(self):
+        from cfme.web_ui.menu import nav
+        nav._nav_to_fn('Compute', 'Infrastructure', 'Resource Pools')(None)
+
+    def resetter(self):
+        # Reset view and selection
+        tb.select("Grid View")
+        sel.check(paginator.check_all())
+        sel.uncheck(paginator.check_all())
+
+
+@navigator.register(ResourcePool, 'Details')
+class Details(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        sel.click(Quadicon(self.obj.name, self.obj.quad_name))
+
+
 def get_all_resourcepools(do_not_navigate=False):
```

Comment:
I'm considering putting these as classmethods in future, then they could pick up the qtype name from the object, what do you think?